publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/UserUploadDetails.java

```diff
@@ -0,0 +1,22 @@
+package no.unit.nva.model.associatedartifacts.file;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.time.Instant;
+import no.unit.nva.commons.json.JsonSerializable;
+import no.unit.nva.model.Username;
+
+@JsonTypeName(UserUploadDetails.TYPE)
+@JsonTypeInfo(use = Id.NAME, property = "type")
+public record UserUploadDetails(Username uploadedBy, Instant uploadedDate) implements UploadDetails, JsonSerializable {
+
+    public static final String TYPE = "UserUploadDetails";
+
+    //TODO: Remove method after migration
+    @JsonProperty
+    public String type() {
+        return TYPE;
+    }
+}
```

Comment:
This method is needed in order to add type property when deserializing. Otherwise we will not be able to use JsonType annotations only after migration.

Reply:
Should there be a test for this behavior? It seems this code can be removed entirely without breaking existing tests, and I don't quite understand what the purpose of it is.

Reply:
That's a good point, adding a test:

```
    @Test
    public void deserializedUploadDetailShouldHaveTypeProperty() throws JsonProcessingException {
        var importUploadDetails = new ImportUploadDetails(null, null, null);
        var userUploadDetails = new UserUploadDetails(null, null);

        var importUploadDetailsJson = dtoObjectMapper.readTree(importUploadDetails.toJsonString());
        var userUploadDetailsJson= dtoObjectMapper.readTree(userUploadDetails.toJsonString());

        assertThat(importUploadDetailsJson.get("type").asText(), is(equalTo(ImportUploadDetails.TYPE)));
        assertThat(userUploadDetailsJson.get("type").asText(), is(equalTo(UserUploadDetails.TYPE)));
    }
```

Reply:
I see! Could still be simplified I think, see Slack explanation.